apps/buildog/src/components/user-auth-form.tsx

```diff
@@ -24,27 +26,47 @@ type LoginFromValues = z.infer<typeof loginSchema>;
 
 export function UserAuthForm({ className, ...props }: UserAuthFormProps) {
   const [loading, setLoading] = React.useState<boolean>(false);
-  const [error, setError] = React.useState<string | null>(null);
   const router = useRouter();
+  const { toast } = useToast();
 
   const form = useForm<LoginFromValues>({
     resolver: zodResolver(loginSchema),
     mode: "onSubmit",
   });
+  //extracts the error code from a firebase error message. => Firebase: Error (auth/invalid-credential)
+  const extractErrorCode = (errorMessage: string): string => {
+    const match = errorMessage.match(/\(auth\/[a-zA-Z0-9\-]+\)/);
+    return match ? match[0].replace(/[()]/g, "") : "unknown-error";
+  };
 
+  // represents the authentication response.
+  interface AuthResponse {
+    email?: string;
+    emailVerified?: boolean;
+    error?: string;
+  }
```

Comment:
Can we move this interface to out of the UserAuthForm component? You can move it on line 19